### PR TITLE
[mod] limiter: minor improvements

### PR DESCRIPTION
**Pull request by return42**
## What does this PR do?

- requests without HTTP header ~`Connection` or missing~ `User-Agent` will be blocked by the limiter

- re_bot is related to `User-Agent` and has been renamed to block_user_agent

---- 

To trace limiter in production I use:

     sudo -H FORCE_SELECTION=Y ./utils/searxng.sh instance inspect | grep limiter

### CODE REVIEW

**Comment by return42** [comment]
With this PR the limiter will block requests where HTTP header `Connection` or `User-Agent` is unset.  I tested this on my instance over a long period without an issue in my (very small) user base.

In order to get a more representative evaluation we are looking out for testers to test this patch in live operation on their SearXNG instances.

Related:  https://docs.searxng.org/admin/installation-nginx.html#nginx-s-searxng-site

**Comment by unixfox** [comment]
Please add me as a reviewer.

File: searx/plugins/limiter.py [hunk 1]
```diff
@@ -62,6 +67,10 @@ def is_accepted_request() -> bool:
             logger.debug("BLOCK %s: got Connection=close", x_forwarded_for)
             return False
 
+        if not request.headers.get('Connection'):
+            logger.debug("BLOCK %s: got Connection=", x_forwarded_for)
```
Comment (mrpaulblack) [inline searx/plugins/limiter.py]: AFAIK the connection header is not used in http2 -> https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection . IMO we should not filter for this header.

**Review by mrpaulblack — CHANGES_REQUESTED** [review]

File: searx/plugins/limiter.py (hunk 1 above)
Comment (unixfox) [reply searx/plugins/limiter.py]: I'm also against this change, I don't think it's wise to block based on this missing HTTP header.

Comment (return42) [reply searx/plugins/limiter.py]: Thanks for having an eye on .. I removed [`if not request.headers.get('Connection'):`](https://github.com/searxng/searxng/compare/553a01171ad48653cf20455cd7f41aa5bba5b662..66810ce71122183f52446cb110dc76e4cae9b7ba).

**Review by unixfox — APPROVED** [review]

**Review by mrpaulblack — APPROVED** [review]
Thanks; LGTM :+1:

**Pull request merged**